src/overlays/actors/ovl_En_Ssh/z_en_ssh.c

```diff
@@ -21,95 +63,798 @@ const ActorInit En_Ssh_InitVars = {
+        return false;
+    } else {
+        return EnSsh_CheckHitBack(this, globalCtx);
+        // Always returns false
+    }
+}
+
+void EnSsh_SetBodyCylinderAC(EnSsh* this, GlobalContext* globalCtx) {
+    Collider_CylinderUpdate(&this->actor, &this->colCylinder[0]);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colCylinder[0].base);
+}
+
+void EnSsh_SetLegsCylinderAC(EnSsh* this, GlobalContext* globalCtx) {
+    s16 angleTowardsLink = ABS((s16)(this->actor.yawTowardsLink - this->actor.shape.rot.y));
+
+    if (angleTowardsLink < 90 * (0x10000 / 360)) {
+        Collider_CylinderUpdate(&this->actor, &this->colCylinder[2]);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colCylinder[2].base);
+    } else {
+        Collider_CylinderUpdate(&this->actor, &this->colCylinder[1]);
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->colCylinder[1].base);
+    }
+}
+
+s32 EnSsh_SetCylinderOC(EnSsh* this, GlobalContext* globalCtx) {
+    Vec3f cyloffsets[] = {
+        { 40.0f, 0.0f, 0.0f },
+        { 0.0f, 0.0f, 0.0f },
+        { -40.0f, 0.0f, 0.0f },
+    };
+    Vec3f cylPos;
+    s32 i;
+
+    for (i = 0; i < 3; i++) {
+        cylPos = this->actor.posRot.pos;
+        cyloffsets[i].x *= this->colliderScale;
+        cyloffsets[i].y *= this->colliderScale;
+        cyloffsets[i].z *= this->colliderScale;
+        Matrix_Push();
+        Matrix_Translate(cylPos.x, cylPos.y, cylPos.z, MTXMODE_NEW);
+        Matrix_RotateY((this->initialYaw / (f32)0x8000) * M_PI, MTXMODE_APPLY);
+        Matrix_MultVec3f(&cyloffsets[i], &cylPos);
+        Matrix_Pull();
+        this->colCylinder[i + 3].dim.pos.x = cylPos.x;
+        this->colCylinder[i + 3].dim.pos.y = cylPos.y;
+        this->colCylinder[i + 3].dim.pos.z = cylPos.z;
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colCylinder[i + 3].base);
+    }
+    return 1;
+}
+
+void EnSsh_SetColliders(EnSsh* this, GlobalContext* globalCtx) {
+    if (this->actor.colChkInfo.health == 0) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->colSph.base);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->colSph.base);
+    } else {
+        if (this->hitTimer == 0) {
+            EnSsh_SetCylinderOC(this, globalCtx);
+        }
+        if (DECR(this->invincibilityTimer) == 0) {
+            EnSsh_SetBodyCylinderAC(this, globalCtx);
+            EnSsh_SetLegsCylinderAC(this, globalCtx);
+        }
+    }
+}
+
+void EnSsh_Init(Actor* thisx, GlobalContext* globalCtx) {
+    f32 frameCount;
+    s32 pad;
+    EnSsh* this = THIS;
+
+    frameCount = SkelAnime_GetFrameCount(&D_06000304.genericHeader);
+    if (this->actor.params == ENSSH_FATHER) {
+        if (gSaveContext.inventory.gsTokens >= 100) {
+            Actor_Kill(&this->actor);
+            return;
+        }
+    } else if (gSaveContext.inventory.gsTokens >= (this->actor.params * 10)) {
```

Comment:
The `EnSshParam` enum should probably not be a thing based on this line. I suggest just using numbers instead, where 0 is the only special value (but doesn't need any name imo).

Reply:
I'm keeping the special value for 0.